### ISSUE TITLE
[casl/ability] Creating complex conditions in Ability Templates

### ISSUE THREAD

**Issue by kshitizshankar** [stalniy/casl#134]
Hey,

I am just getting started with Casl and trying to use it in a microservices architecture through JWT Tokens. 

Here's the context of what I am trying to do: 
1. The Ability Template is defined through an Admin UI with variables. (There is a default one)
2. The Ability gets loaded and is processed (Awesome documentation around it already exists so thanks!)

The key question that I am struggling a bit with - How do I create an ability template which has multiple conditions that might require a bit more complex logic?

For example - 
  ```
  {
        //Any member can manage his/her profile
        "actions": ["read", "create", "update", "delete"],
        "subject": "UserProfiles",
        "conditions": {
            "id": "${user.id}"
        }
    },
    {
        //Any member can create a project
        "actions": ["create"],
        "subject": "Projects",
        "conditions": {}
    },
    **{
        //Admin Project members can manage their projects given both are part of the same organization.
        "actions": ["read", "create", "update", "delete"],
        "subject": "Projects",
        "conditions": {
            //How do I crate a condition that the 
            //${project.admins[]} contains $(user.id)
            // ${project.organizations[]} contains ${user.org}
        }
    },**
```
How do I create complex conditions? Most probably I am missing something obvious here.

Thanks!

**Comment by carlespibernat** [comment]
Hi! How did you solved this?

**Comment by kshitizshankar** [comment]
@carlespibernat Sorry but this was a long time back and I don't really remember the context. I was playing around with CASL at that time and I have no idea why I closed this issue without adding anything else.

I don't have this approach anymore (so I am guessing I dropped this line of thinking) and am not really storing the templates as such. I create the abilities on the fly with the populated variables from the user on the server. With that, I didn't have the need to store anything in a template file and just using normal functions to populate the rules on the fly.
